src/Common/tests/System/Net/Sockets/SocketTestExtensions.cs

```diff
@@ -23,5 +23,15 @@ public static TcpListener CreateAndStartTcpListenerOnAnonymousPort(out int port)
             port = ((IPEndPoint)listener.LocalEndpoint).Port;
             return listener;
         }
+
+        // On non-Windows platforms, once non-blocking is turned on (either explicitly
+        // or by performing an async operation), we always stay in non-blocking mode.
+        // Therefore, sync operation have to be simulated via async and explicit blocking.
+        // Force us into this mode for testing purposes.
+        public static void ForceNonBlocking(this Socket socket)
```

Comment:
Since most of the call sites are guarded by a condition, how about adding a bool arg to this method and moving the condition to in here?